ckks/bootstrapping/default_params.go

```diff
@@ -24,674 +22,174 @@ var DefaultParametersDense = []defaultParametersLiteral{N16QP1767H32768H32, N16Q
 var (
 	// N16QP1546H192H32 is a default bootstrapping parameters for a main secret with H=192 and an ephemeral secret with H=32.
-	// Residual Q : 420 bits.
+	// Residual Q : []int{60, 40, 40, 40, 40, 40, 40, 40, 40, 40} (420 bits).
+	// SlotsToCoeffs Q: []int{39, 39, 39}.
+	// EvalMod Q: []int{60, 60, 60, 60, 60, 60, 60, 60}.
+	// CoeffsToSlots Q: []int{56, 56, 56, 56}.
 	// Precision : 26.6 bits for 2^{15} slots.
 	// Failure : 2^{-138.7} for 2^{15} slots.
 	N16QP1546H192H32 = defaultParametersLiteral{
 		ckks.ParametersLiteral{
-			LogN:  16,
-			Sigma: rlwe.DefaultSigma,
-			H:     192,
-			Q: []uint64{
-				0x10000000006e0001, // 60 Q0
-				0x10000140001,      // 40
-				0xffffe80001,       // 40
-				0xffffc40001,       // 40
-				0x100003e0001,      // 40
-				0xffffb20001,       // 40
-				0x10000500001,      // 40
-				0xffff940001,       // 40
-				0xffff8a0001,       // 40
-				0xffff820001,       // 40
-				0x7fffe60001,       // 39 StC
-				0x7fffe40001,       // 39 StC
-				0x7fffe00001,       // 39 StC
-				0xfffffffff840001,  // 60 Sine (double angle)
-				0x1000000000860001, // 60 Sine (double angle)
-				0xfffffffff6a0001,  // 60 Sine
-				0x1000000000980001, // 60 Sine
-				0xfffffffff5a0001,  // 60 Sine
-				0x1000000000b00001, // 60 Sine
-				0x1000000000ce0001, // 60 Sine
-				0xfffffffff2a0001,  // 60 Sine
-				0x100000000060001,  // 56 CtS
-				0xfffffffff00001,   // 56 CtS
-				0xffffffffd80001,   // 56 CtS
-				0x1000000002a0001,  // 56 CtS
-			},
-			P: []uint64{
-				0x1fffffffffe00001, // Pi 61
-				0x1fffffffffc80001, // Pi 61
-				0x1fffffffffb40001, // Pi 61
-				0x1fffffffff500001, // Pi 61
-				0x1fffffffff420001, // Pi 61
-			},
-			LogSlots:     15,
-			DefaultScale: 1 << 40,
-		},
-
-		Parameters{
-			EphemeralSecretWeight: 32,
-			SlotsToCoeffsParameters: advanced.EncodingMatrixLiteral{
-				LinearTransformType: advanced.SlotsToCoeffs,
-				RepackImag2Real:     true,
-				LevelStart:          12,
-				BSGSRatio:           2.0,
-				BitReversed:         false,
-				ScalingFactor: [][]float64{
-					{0x7fffe60001},
-					{0x7fffe40001},
-					{0x7fffe00001},
-				},
-			},
-			EvalModParameters: advanced.EvalModLiteral{
-				Q:             0x10000000006e0001,
-				LevelStart:    20,
-				SineType:      advanced.Cos1,
-				MessageRatio:  256.0,
-				K:             16,
-				SineDeg:       30,
-				DoubleAngle:   3,
-				ArcSineDeg:    0,
-				ScalingFactor: 1 << 60,
-			},
-			CoeffsToSlotsParameters: advanced.EncodingMatrixLiteral{
-				LinearTransformType: advanced.CoeffsToSlots,
-				RepackImag2Real:     true,
-				LevelStart:          24,
-				BSGSRatio:           2.0,
-				BitReversed:         false,
-				ScalingFactor: [][]float64{
-					{0x100000000060001},
-					{0xfffffffff00001},
-					{0xffffffffd80001},
-					{0x1000000002a0001},
-				},
-			},
+			LogN:     16,
+			LogQ:     []int{60, 40, 40, 40, 40, 40, 40, 40, 40, 40},
+			LogP:     []int{61, 61, 61, 61, 61},
+			H:        192,
+			LogScale: 40,
```

Comment:
Is `LogScale` ever not equal to the small moduli in `LogQ` (excluding the first one)? I don't think being able to create these "raw" `ParametersLiteral` like this is safe because type invariants are not checked. Please see my comment on using a "constructor" above.

Reply:
`ParametersLiteral` are checked by the constructor `NewParametersFromLiteral`. We want to keep the maximum possible flexibility for the user uppon defining the `ParametersLiteral`, such as being able to specify the size, or even the prime, of the moduli chain. This is required for circuit optimization.

Reply:
Got it. Thank you!